### PR TITLE
Fix get_salon_names request bug

### PR DESCRIPTION
**Pull request by kdious**
Made it so that this uses the Twisted deferred request and handles
the call back properly.  Now, the list of salon names will be
returned as a JSON array in the response boody.

### CODE REVIEW

**Comment by kdious** [comment]
Confirmed that this works:

1. Used sqlite3 to insert a row into the salons table
```
.open /home/ubuntu/src/harold/test.db

insert into salons (name, conch_emoji, allow_deploys, deploy_hours_start, deploy_hours_end, tz) values ('handbook', ':book:', 0, '0500', '1700', 'America/Los_Angeles');
```

2. Did a request to Harold
```
ubuntu@ip-172-31-19-178:~$ python
Python 2.7.12 (default, Aug 22 2019, 16:36:40) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import requests
>>> import hmac
>>> import hashlib
>>> import urllib
>>> import urlparse
>>> 
>>> secret = 'asupersecrettoken'
>>> data = {'reason': 'Test', 'salon-name': 'test-salon'}
>>> body = urllib.urlencode(data)
>>> hash = hmac.new(secret, body, hashlib.sha1)
>>> headers = {"Content-Type": "application/x-www-form-urlencoded", "X-Hub-Signature": "sha1=" + hash.hexdigest()}
>>> 
>>> url = 'http://0.0.0.0:8080/harold/deploy/get_salon_names'
>>> resp = requests.get(url, headers=headers, data=body)
>>> resp.json()
[u'handbook']
>>> 
```

**Comment by kdious** [comment]
👓 @spladug @andytuba 

Had to fix this bug so that I can fetch the salon names properly.

File: harold/plugins/deploy.py [hunk 1]
```diff
@@ -217,13 +217,19 @@ class DeployGetSalonNamesListener(DeployListener):
...
 
     def _handle_request(self, request):
-        salons = yield self.salons.all()
-        salon_names = [salon.name for salon in salons]
 
-        # Configure the response
-        request.setHeader("Content-Type", "application/json")
-        request.write(json.dumps(salon_names))
-        request.finish()
+        def send_response(salons):
```
Comment (kdious) [inline harold/plugins/deploy.py]: Inspiration:
https://github.com/spladug/harold/blob/c46487a1db6f25d7c0bc7ddb59823e4766166e9e/harold/plugins/deploy.py#L89

File: harold/plugins/http.py [hunk 2]
```diff
@@ -46,14 +46,15 @@ def _authenticate_request(self, request):
...
 
     def render_GET(self, request):
+        response = ""
         try:
             self._authenticate_request(request)
         except AuthenticationError:
             request.setResponseCode(403)
         else:
-            self._handle_request(request)
+            response = self._handle_request(request)
```
Comment (kdious) [inline harold/plugins/http.py]: Need to make sure we're sure to return the `NOT_DONE_YET` response:
https://github.com/spladug/harold/pull/42/files#diff-270925a10931db85f0b8727a98cef1edR232

File: harold/plugins/http.py [hunk 3]
```diff
@@ -46,14 +46,15 @@ def _authenticate_request(self, request):
...
 
-        return ""
+        return response
```
Comment (spladug) [inline harold/plugins/http.py]: should this be `return response or ""` to deal with the other handlers which are just returning `None` by default?

**Comment by kdious** [comment]
💇 Good idea @spladug

**Pull request merged**